chia/wallet/nft_wallet/nft_puzzles.py

```diff
@@ -142,7 +142,7 @@ def get_nft_info_from_puzzle(puzzle: Program, nft_coin: Coin) -> NFTInfo:
         uncurried_nft.owner_did.as_python(),
         uint64(uncurried_nft.trade_price_percentage.as_int()),
         data_uris,
-        uncurried_nft.data_hash.as_python().hex(),
+        str(uncurried_nft.data_hash.as_python(), "utf-8"),
```

Comment:
if the data is not valid utf-8, this would throw. There's no test covering this case. (but I think this change is poorly justified to begin with)